### PR TITLE
Association

### PR DESCRIPTION
**Pull request by Therealadetunji**
Create a class Classroom with the following:
@label instance variable, should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor). ✔️ 

Create a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor). ✔️ 

Create a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor). ✔️

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class Classroom
+  def initialize(label, _students)
+    @label = label
+    @_students = []
+  end
+  attr_accessor :label
+
+  def add_students(student)
+    @students.push(student)
+  end
```
Comment (1GORDON) [inline classroom.rb]: Great job @Therealadetunji 🤝

- [ ] Kindly make sure this method serves as the `has-many` side (a classroom has many students). You can check this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) for more information.
- [ ] You may also add the attribute reader for students `attr_reader :students`.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class Person < Nameable
+  def initialize(age, name = 'Unknown', _parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @_parent_permission = parent_permission
+    @visits =  []
+  end
```
Comment (1GORDON) [inline person.rb]: Great job @Therealadetunji 🤝

- [ ] Kindly initialize @rentals. Please check this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) for more guidance.
- [ ] You may also add a method for adding rentals. Kindly check this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) for more guidance.

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Therealadetunji ✋

Good job so far!

## Highlights!
- Followed Gitflow ✔️ 
- Descriptive Pull request ✔️
- Professional readme ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️

- [ ] Kindly make sure the linter checks are passing and in green. This is a critical requirement for the approval of your project.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
+require './nameable'
+
+class Person < Nameable
+  def initialize(age, name = 'Unknown', _parent_permission: true)
+    super()
+    @id = Random.rand(1..1000)
```
Comment (tufoinnkuo10) [inline person.rb:6]: - [ ] Great job. please like the previous reviewer noted, Kindly initialize @rentals. Please check this [link](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/class_and_object_for_irb.md#constructor-and-instance-vars) for more guidance.

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Therealadetunji 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally

Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: person.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
+require './nameable'
+
+class Person < Nameable
+  def initialize(age, rentals, name = 'Unknown', _parent_permission: true)
```
Comment (GzimAsani) [inline person.rb:4]: - [ OPTIONAL ] rentals does not need to be included here, it just needs to be initialized down there as you already did

File: person.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @_parent_permission = parent_permission
+    @rentals = rentals
+    @visits = []
+  end
+  attr_reader :id, :rentals
+  attr_accessor :name, :age
```
Comment (GzimAsani) [inline person.rb:14]: - [ OPTIONAL ] Also rental should be put in attr_accessor since we need it as getter an setter

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## There is nothing else to say than it's time to merge... :partying_face: 
Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


## Optional suggestions
-N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Kindly check the comment below

### Please be reminded to not open a new pull request after you get a review, only use the same one :thought_balloon: 


 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**